bin/check-dependency-version-consistency.ts

```diff
@@ -32,38 +32,49 @@ function collect(value: string, previous: string[]) {
 }
 
 // Setup CLI.
-const program = new Command();
-program
-  .version(getCurrentPackageVersion())
-  .argument('<path>', 'path to workspace root')
-  .option(
-    '--fix',
-    'Whether to autofix inconsistencies (using highest version present)',
-    false
-  )
-  .option(
-    '--ignore-dep <dependency>',
-    'Dependency to ignore (option can be repeated)',
-    collect,
-    []
-  )
-  .action(function (path, options: { ignoreDep: string[]; fix: boolean }) {
-    // Calculate.
-    const dependencyVersions = calculateVersionsForEachDependency(path);
-    let mismatchingVersions = filterOutIgnoredDependencies(
-      calculateMismatchingVersions(dependencyVersions),
-      options.ignoreDep
-    );
+function run() {
+  const program = new Command();
 
-    if (options.fix) {
-      mismatchingVersions = fixMismatchingVersions(path, mismatchingVersions);
-    }
+  program
+    .version(getCurrentPackageVersion())
+    .argument('<path>', 'path to workspace root')
+    .option(
+      '--fix',
+      'Whether to autofix inconsistencies (using highest version present)',
+      false
+    )
+    .option(
+      '--ignore-dep <dependency>',
+      'Dependency to ignore (option can be repeated)',
+      collect,
+      []
+    )
+    .action(function (path, options: { ignoreDep: string[]; fix: boolean }) {
+      // Calculate.
+      const dependencyVersions = calculateVersionsForEachDependency(path);
+      let mismatchingVersions = filterOutIgnoredDependencies(
+        calculateMismatchingVersions(dependencyVersions),
+        options.ignoreDep
+      );
 
-    // Show output.
-    if (mismatchingVersions.length > 0) {
-      const outputLines = mismatchingVersionsToOutputLines(mismatchingVersions);
-      outputLines.forEach((line) => console.log(line));
-      process.exitCode = 1;
-    }
-  })
-  .parse(process.argv);
+      if (options.fix) {
+        mismatchingVersions = fixMismatchingVersions(path, mismatchingVersions);
+      }
+
+      // Show output.
+      if (mismatchingVersions.length > 0) {
+        const outputLines =
+          mismatchingVersionsToOutputLines(mismatchingVersions);
+        outputLines.forEach((line) => console.log(line));
+        process.exitCode = 1;
+      }
+    })
+    .parse(process.argv);
+}
+
+try {
+  run();
+} catch (e) {
+  console.error((e as Error).message);
```

Comment:
Was looking at how to handle the error type in TypeScript. I think this would be better:
```suggestion
  if (e instanceof Error) {
    console.error(e.message);
  }
```
https://fettblog.eu/typescript-typing-catch-clauses/

Reply:
👍 We still need the `process.exitCode = 1` here because the exit code is `0` without it.